SYCL/OptionalKernelFeatures/is_compatible.cpp

```diff
@@ -42,5 +46,31 @@ int main() {
     Called = true;
   }
 
+  if (sycl::is_compatible<GoodWGSize>(Dev)) {
+    Q.submit(
+        [&](sycl::handler &h) { 
+          h.parallel_for<class GoodWGSize>(sycl::range<2>(4, 2),
+                   [=](sycl::item<2> it) [[sycl::reqd_work_group_size(SIZE, SIZE)]] {});
+        });
+    Q.wait();
+    Compatible &= (Dev.get_info<sycl::info::device::max_work_group_size>() > (SIZE + SIZE));
+    Called = true;
+  }
+
+  if (sycl::is_compatible<WrongReqWGSize>(Dev)) {
+    Q.submit(
+        [&](sycl::handler &h) { 
+          h.parallel_for<class WrongReqWGSize>(sycl::range<1>(2),
+                   [=](sycl::item<1> it) [[sycl::reqd_work_group_size(INT_MAX)]] {});
+        });
+    Q.wait();
+    Compatible &= false;
+    Called = false;
+  }
+  else {
+    Compatible &= true;
+    Called = true;
+  }
```

Comment:
I would rather check if `INT_MAX` exceeds `device::max_work_group_size` and that's true, then assert that `is_compatible<WrongReqWGSize>` returns `false`